resource/item.go

```diff
@@ -24,11 +24,16 @@ type Item struct {
 
 // ItemList represents a list of items
 type ItemList struct {
+	// Results defines the total number of items in the collection matching the current
+	// context without any filter applied. If the storage handler cannot compute this value, -1 is set.
+	Results int
```

Comment:
This chunk is not part of offset support and should be removed/proposed as a separate PR (I'm not sure to understand the use-case though).